lighthouse-core/report/templates/report-template.html

```diff
@@ -177,3 +176,13 @@ <h1 class="menu__header-title">Lighthouse</h1>
     </footer>
   </section>
 </div>
+
+{{#if scripts }}
+<script>
+  window.lhresults = {{{lhresults}}};
```

Comment:
> Really we should be doing more to escape our JS, but one end run around this is to do double stash` {{lhresults}}` in a (display: none) div, so we get HTML entity escaping, then get the actual results when needed by JSON.parse(lhresultsDiv.textContent) (see rule 1 and the summary section for more details)

I'm not sure this does anything different other than adding an additional parse. Handlebars, https://github.com/GoogleChrome/lighthouse/issues/1188, and https://github.com/GoogleChrome/lighthouse/issues/996 already force entity escaping everywhere that's needed.

The `lhresults` object is also constructed and sanitized by us. It's not pure user-input. Parts of the object come from user-space but we only allow markdown (two tags at that). If the user provides HTML tags, they'll end up encoded.

> This suggestion is actually independent of the postMessage part of this PR, and just about (safely) embedding lhresults in the page again. 

The thing we're trying to prevent is a malicious actor creating gists in a user's account. That all happens online. Locally, I'm not sure what there is to worry about? A malicious actor would have to create a page such that running LH over it properly escapes the embedded JSON (which we prevent), alters `lhresults` to their advantage, the user clicks "share in viewer",  the rev'd `lhresults` is able to escape the online viewer's `{{}}`, passes additional `JSON.parse()` and DOM construction calls, then has code to automatically call `createGist()` ([we don't provide a way to call this](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-viewer/app/src/main.js#L44); creating a gist is behind user gesture atm). The also needs to already have an oauth token. 

If something were to ahppen, our oauth tokens are limited scoped. They can only create gists in the users account. They're also secret by default (for extra feel goods) and can be revoked by the user at any time.

The theme of this PR is to encourage users to share reports.  Given the current low volume of sharers, let's land this so we can grow some us LH users!